src/esp/assets/ResourceManager.cpp

```diff
@@ -2159,87 +2312,185 @@ void ResourceManager::loadMeshes(Importer& importer,
   }
 }  // ResourceManager::loadMeshes
 
+Mn::Image2D ResourceManager::convertRGBToSemanticId(
+    const Mn::ImageView2D& srcImage,
+    Cr::Containers::Array<Mn::UnsignedShort>& clrToSemanticId) {
+  // convert image to semantic image here
+
+  const Mn::Vector2i size = srcImage.size();
+  // construct empty integer image
+  Mn::Image2D resImage{
+      Mn::PixelFormat::R16UI, size,
+      Cr::Containers::Array<char>{
+          Mn::NoInit,
+          std::size_t(size.product() * pixelSize(Mn::PixelFormat::R16UI))}};
+
+  Cr::Containers::StridedArrayView2D<const Mn::Color3ub> input =
+      srcImage.pixels<Mn::Color3ub>();
+  Cr::Containers::StridedArrayView2D<Mn::UnsignedShort> output =
+      resImage.pixels<Mn::UnsignedShort>();
+  for (std::size_t y = 0; y != size.y(); ++y) {
+    Cr::Containers::StridedArrayView1D<const Mn::Color3ub> inputRow = input[y];
+    Cr::Containers::StridedArrayView1D<Mn::UnsignedShort> outputRow = output[y];
+    for (std::size_t x = 0; x != size.x(); ++x) {
+      /* Fugly. Sorry. Needs better API on Magnum side. */
+      const Mn::Color3ub color = inputRow[x];
+
+      const Mn::UnsignedInt colorInt =
+          color.r() << 16 | color.g() << 8 | color.b();
+      outputRow[x] = clrToSemanticId[colorInt];
+    }
+  }
+  // upload the `integer` image to an integer Texture2D and then use as
+  // a semantic texture in the shader
```

Comment:
:eyes: ... feels like these `//` comments should have been removed from this function, it makes no sense here

Same below, heh.